src/wobbly/service.py

```diff
@@ -112,6 +112,35 @@ async def delete(self, job_id: JobIdentifier) -> None:
             job=job_id.id,
         )
 
+    async def delete_expired(self) -> None:
+        """Delete all jobs that have passed their destruction time.
+
+        The jobs are deleted out of the database entirely, not moved to
+        ``ARCHIVED`` status.
+
+        Be aware that Wobbly has no access to the job queue and therefore
+        cannot cancel deleted jobs, so if the destruction time is very short,
+        the job may still be executing, and the attempt to update the record
+        when it completes will fail with an HTTP 404 error.
+        """
+        jobs = await self._storage.list_expired()
+        if jobs:
+            self._logger.info(f"Deleting {len(jobs)} expired jobs")
+        count = 0
+        for job in jobs:
+            job_id = JobIdentifier(
+                service=job.service, owner=job.owner, id=job.id
+            )
+            if await self._storage.delete(job_id):
```

Comment:
Would we ever have enough expired jobs that we'd stress the DB too much by running a query for each one, vs deleting them all in one query? I get that we're doing them one by one here at least in part to get useful loggin.

Reply:
Yeah, good call, I was mostly being lazy. I'll add a bulk delete method to the storage layer and use that instead.